hack/github-oidc-setup.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
```

Comment:
this script is needed? i did not see that being used in the release action

Reply:
apologies for not referencing in the README. 
it is needed to setup the oidc access for github actions to authenticate against GCP to kick off the gcloud builds command.

updated README.md to explain it is a one time setup.